### DISCUSSION TITLE
[Question] Local-only records?

### DISCUSSION THREAD

**Discussion by ZimM-LostPolygon** [electric-sql/electric#1193] [Feature requests]
Hello! Here's my use-case: 
I'm writing a note-taking app with cloud sync (think EverNote, OneNote etc.), and ElectricSQL looks like a great fit for that. However, for extra-sensitive data, the user might choose to NOT sync specific notes and only keep them on the device the note is created on.

Is there a way to achieve this with ElectricSQL? Ideally, the API for queries wouldn't change. Local-only notes will still be grabbed from local SQLite database, just ignored when it comes to sync if a special flag is set.

Thank you!

**Comment by linear** [comment]
<p><a href="https://linear.app/electric-sql/issue/VAX-1805/[question]-local-only-records">VAX-1805 [Question] Local-only records?</a></p>

**Comment by alco** [comment]
Hey @ZimM-LostPolygon. I have converted this from an issue to a discussion.

We don't currently have this feature implemented. As a workaround, you can create a separate local table by using the SQLite adapter API directly. Electric won't know anything about it but your client app will need to know to query from both the electrified and the local tables, as well as which notes to write where.

**Reply by ZimM-LostPolygon** [reply]
This basically triples the amount of actions required, so not a real solution for me, unfortunately. Imagine the simple task of retrieving a list of notes ordered by the creation date: 

1) Get an ordered list of cloud notes from Electric 
2) Get an ordered list of local-only nodes from SQLite
3) Manually merge two lists and then sort the resulting list again on the client

At that point, it's usually just not worth it :(

That said, I understand that it's not possible right now with just Electric, but do you think maybe this is something that might be supported in the future?

Thank you!

**Reply by thruflo** [reply]
Could you not just join in the query, so e.g.: do a FULL OUTER JOIN across the two tables in your liveQuery?

The data synced via Electric is just stored in the local DB, so you can have the database engine do the sorting for you and you can use raw SQL in a live query.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
